plugins/cake/cake_tin_delay_

```diff
@@ -0,0 +1,377 @@
+#!/bin/bash
+# -*- sh -*-
+
+: << =cut
+
+=pod
+
+=encoding UTF-8
+
+=head1 NAME
+
+cake_tin_delay_ - Plugin to monitor cake's average tin delay 
+
+=head1 CONFIGURATION
+
+None needed.
+
+=head1 INTERPRETATION
+
+Cake, also known as sch_cake is a modern bandwidth limiter, which eliminates
+buffer bloat over slow links. It's also capable to give flows, hosts and each 
+flow of each host a fair part of the avaible bandwidth.
+
+This plugin allows for a monitoring packet delays per tins on the qdisc.
+
+=head1 SEE ALSO
+
+Take a look at "man cake" to get more information about cake.
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf suggest
+
+=head1 AUTHORS
+
+RubenKelevra <ruben@vfn-nrw.de>
+
+work based on the tc plugin, authors:
+Steve Schnepp <steve.schnepp@gmail.com>,
+Samuel Smith <esaym@cpan.org>,
+Nye Liu <nyet@nyet.org>
+
+=head1 LICENSE
+
+GPLv2 or later
+
+=cut
+
+DEVICE=${0##*/cake_tin_delay_}
+
+tc_get_ifb_dev() {
+	dev="$(/sbin/tc filter show dev $DEVICE parent ffff: protocol all | grep "mirred" | grep "Egress Redirect to device")"
+	[ $? -eq 1 ] && return 1
+	
+	echo "$dev" | sed -n 's/.* device \([^ ]*\).*/\1/p' | tr ')' ' ' | awk '{ print $1 }'
+}
+
+tc_cake_get_diffserv() {
+	/sbin/tc -s qdisc show dev "$1" | head -n1 | grep -i "diffserv3" > /dev/null
+	if [ $? -eq 0 ]; then
+		echo 3
+	fi
+	/sbin/tc -s qdisc show dev "$1" | head -n1 | grep -i "diffserv4" > /dev/null
+	if [ $? -eq 0 ]; then
+		echo 4
+	fi
+	/sbin/tc -s qdisc show dev "$1" | head -n1 | grep -i "diffserv8" > /dev/null
+	if [ $? -eq 0 ]; then
+		echo 8
+	fi
+	return 0
+}
+
+convert_time() {
+	#converts us, ms, s into s without a unit
+	echo $1 | awk '{
+		time = $1;
+		sub(/[a-z]+$/, "", time);
+		unit = $1;
+		sub(/^[^a-z]+/, "", unit);
+
+		if (unit == "ns") {
+			time /= 1000000000;
+		} else if (unit == "us") {
+			time /= 1000000;
+		} else if (unit == "ms") {
+			time /= 1000;
+		} else if (unit == "s") {
+			time = time;
+		} else if (unit == "m") {
+			time *= 60;
+		} else if (unit == "h") {
+			time *= 3600;
+		} else if (unit == "d") {
+			time *= 86400;
+		}
+		printf("%.9f\n", time)
+	}'
+}
+
+tc_cake_get_delays() {
+	delays="$(/sbin/tc -s qdisc show dev "$1" | grep "^  av_delay" | sed -e 's/av_delay//')"
+	for e in ${delays[@]}; do
+		delay="$(convert_time $e)"
+		echo -en " $delay "
```

Comment:
What is the purpose of `-e` here?
Regarding `-n`: "printf" should be more portable (in case you switch from bash to sh).